app/src/main/java/ml/docilealligator/infinityforreddit/activities/ViewPostDetailActivity.java

```diff
@@ -204,72 +245,101 @@ protected void onCreate(Bundle savedInstanceState) {
-                }
-                adjustToolbar(mToolbar);
-
-                int navBarHeight = getNavBarHeight();
-                if (navBarHeight > 0) {
-                    CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) fab.getLayoutParams();
-                    params.bottomMargin += navBarHeight;
-                    fab.setLayoutParams(params);
-
-                    searchPanelMaterialCardView.setContentPadding(searchPanelMaterialCardView.getPaddingStart(),
-                            searchPanelMaterialCardView.getPaddingTop(),
-                            searchPanelMaterialCardView.getPaddingEnd(),
-                            searchPanelMaterialCardView.getPaddingBottom() + navBarHeight);
-                }
-            }
+    private void setupWindow() {
+        if (isChangeStatusBarIconColor()) {
+            addOnOffsetChangedListener(mAppBarLayout);
         }
 
-        boolean swipeBetweenPosts = mSharedPreferences.getBoolean(SharedPreferencesUtils.SWIPE_BETWEEN_POSTS, false);
-        if (!swipeBetweenPosts) {
-            if (mSharedPreferences.getBoolean(SharedPreferencesUtils.SWIPE_RIGHT_TO_GO_BACK, true)) {
-                mSlidrInterface = Slidr.attach(this);
-            }
-            viewPager2.setUserInputEnabled(false);
+        if (isImmersiveInterface()) {
+            setupImmersiveInterface();
+        }
+    }
+
+    private void setupImmersiveInterface() {
+        Window window = getWindow();
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            window.setDecorFitsSystemWindows(false);
+        } else {
+            window.setFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS, WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS);
+        }
+        adjustToolbar(mToolbar);
+
+        int navBarHeight = getNavBarHeight();
+        if (navBarHeight > 0) {
+            CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) fab.getLayoutParams();
+            params.bottomMargin += navBarHeight;
+            fab.setLayoutParams(params);
+
+            searchPanelMaterialCardView.setContentPadding(searchPanelMaterialCardView.getPaddingStart(),
+                    searchPanelMaterialCardView.getPaddingTop(),
+                    searchPanelMaterialCardView.getPaddingEnd(),
+                    searchPanelMaterialCardView.getPaddingBottom() + navBarHeight);
         }
+    }
+
+    private void setupSwipeRightOutOfPost() {
+        if (mSharedPreferences.getBoolean(SharedPreferencesUtils.SWIPE_RIGHT_TO_GO_BACK, true)) {
+            mSlidrInterface = Slidr.attach(this);
+        }
+        viewPager2.setUserInputEnabled(false);
+    }
+
+    private void setupSwipeBetweenPosts() {
         postFragmentId = getIntent().getLongExtra(EXTRA_POST_FRAGMENT_ID, -1);
-        if (swipeBetweenPosts && posts == null && postFragmentId > 0) {
```

Comment:
Why remove `swipeBetweenPosts`?

Reply:
This method is only called when `swipeBetweenPosts` is `true` thanks to the `shouldSwipeBetweenPosts()` method.

Dropping the relevant code for your reference:

https://github.com/Docile-Alligator/Infinity-For-Reddit/blob/0320aed162c1aecc113099c959b2b92b96c63302/app/src/main/java/ml/docilealligator/infinityforreddit/activities/ViewPostDetailActivity.java#L203-L207

https://github.com/Docile-Alligator/Infinity-For-Reddit/blob/0320aed162c1aecc113099c959b2b92b96c63302/app/src/main/java/ml/docilealligator/infinityforreddit/activities/ViewPostDetailActivity.java#L295-L311